### ISSUE TITLE
Plex is not up to date, package is too large

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#11661]
We're working to update how Plex is published - https://github.com/IBM/plex/discussions/452. It's likely Plex will start to publish individual packages per-family.

Currently we're blocked by this. The compressed package binary is too large, https://github.com/IBM/plex/issues/453. Once this has been addressed, we need to update packages:

**Checklist**

- [x] Update Plex package(s) and versions
- [x] Determine how we would need to revise the sass font resolver, and then update it
- [x] Ensure no regressions
- [ ] Put together an e2e/percy snapshot for the Plex families Carbon includes (nice to have)

**Workaround**

To use the most up to date Plex assets with Carbon, you can configure the sass to use the Akamai server. It can be configured via the `$use-akamai-cdn` scss config variable.

**Comment by mjabbink** [comment]
@tay1orjones @alisonjoseph Making the note here that the “ﬁ” ligature should be the default setting. Whatever Carbon for IBM.com is doing and what Carbon did with v10 is the correct expectation.

**Comment by sstrubberg** [comment]
@tay1orjones can you remind me where we're at with this? Is this the one we're punting to Brand?

**Comment by tay1orjones** [comment]
@sstrubberg This issue is for the work we need to do in this `carbon` repo, no matter who updates the Plex repo. We can't do this work until the Plex repo is updated and publishes an npm package for each Plex font family.

**Comment by tay1orjones** [comment]
Plex has been coverted to a monorepo, https://github.com/IBM/plex/pull/577, and is now publishing a collection of per-family packages. These packages should be individually small enough to work around this issue.

We need to update the default font resolver to work with the new packages and directory structure.

**Comment by tay1orjones** [comment]
Closed via https://github.com/carbon-design-system/carbon/pull/17025

Thanks again @tw15egan! 🎉

**Comment by tay1orjones** [comment]
The original fix was reverted in https://github.com/carbon-design-system/carbon/pull/17211, we need to revisit this and determine a way to introduce this change without breaking projects that are already configuring `$font-path`

**Comment by s100** [comment]
I'm interested in seeing this fixed. Currently our application has several direct and indirect dependencies which pull in `@ibm/plex@6.4.1`, meanwhile we also use `@carbon/styles` which pulls in `@ibm/plex@6.0.0-next.6`. The IBM Plex font package is large in itself, but having two of them is quite inconvenient.

**Comment by tay1orjones** [comment]
The approach we're circling around for the second rev of this is to introduce a new boolean config variable (false by default) that allows users to opt into the new pathing. The paths can still use the existing `$font-path` variable, just modify the `/` to be `-{familyName}/`. We considered deprecating `$font-path`, but that's not necessary.

This approach will result in essentially the same PR that @tw15egan already did, but with 
* the addition of the new variable 
* making sure we don't uninstall `@ibm/plex` as it'll still be needed for the "legacy" pathing

Some new docs around this change would be good to have as well. I think there's a few places `$font-path` is used/documented.
